dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java

```diff
@@ -49,12 +49,27 @@ public List<Event> findEventsForOrganization(long organizationId, Integer offset
         return list(query);
     }
 
-    public List<Event> findEventsForInitiatorUser(long initiatorUser, Integer offset, Integer limit) {
+    public List<Event> findEventsForInitiatorUser(User loggedInUser, long initiatorUser, Integer offset, Integer limit) {
         Query<Event> query = namedTypedQuery("io.dockstore.webservice.core.Event.findAllByInitiatorUserId")
-            .setParameter("initiatorUser", initiatorUser)
-            .setFirstResult(offset)
-            .setMaxResults(limit);
-        return list(query);
+            .setParameter("initiatorUser", initiatorUser);
+
+        //when viewing your own profile, show all events
+        if (loggedInUser.getId() == initiatorUser) {
+            query.setFirstResult(offset).setMaxResults(limit);
+            return list(query);
+        }
+        List<Event> filteredEvents = new ArrayList<>();
+        //First, filters out all category events. Then filters out events regarding unpublished entries when viewing other users' profiles (not your own profile)
+        filterCategoryEvents(loggedInUser, query.getResultList()).forEach(event -> {
+            if (event.getWorkflow() == null && event.getTool() == null) {
```

Comment:
This filter might need to check `AppTools` and `Notebooks`, also.  Wonder if it makes sense to add a `getEntry` method to `Event`?  Then, we could replace the multiple `ifs` with something like:
```
   if (event.getEntry() != null && event.getEntry().isPublished()) {
       filteredEvents.add(event);
   }
```

Reply:
From discussion with @kathy-t, we decided it would be best to create a seperate ticket to filter out events for unpublished workflows. So this filter is removed from this PR. I also think it would be great to add a `getEntry` method to `Event` possibly when working on the new [ticket](https://ucsc-cgl.atlassian.net/browse/SEAB-5422)